lib/src/components/NxCollapsibleItemsSelect/NxCollapsibleMultiSelect/__tests__/NxCollapsibleMultiSelect.test.tsx

```diff
@@ -4,355 +4,559 @@
+      expect(view.container).toContainElement(childrenEl);
+      expect(childrenEl).toContainElement(childEl);
     });
 
-    it('renders unchecked <NxCheckbox> for all options if selectedIds prop is not provided', function () {
-      const shallowRender = getShallow();
-      const renderOption = shallowRender.renderProp('renderOption');
+    it('sets aria-expanded iff isOpen prop is true and there are options', function() {
+      expect(renderAndGetTrigger()).toHaveAttribute('aria-expanded', 'false');
+      expect(renderAndGetTrigger({ options: [] })).toHaveAttribute('aria-expanded', 'false');
+      expect(renderAndGetTrigger({ isOpen: true })).toHaveAttribute('aria-expanded', 'true');
+    });
+
+    it('sets disabled if the disabled prop is true or there are no options', function() {
+      expect(renderAndGetTrigger()).toBeEnabled();
+      expect(renderAndGetTrigger({ disabled: true })).toBeDisabled();
+      expect(renderAndGetTrigger({ disabled: null })).toBeEnabled();
+      expect(renderAndGetTrigger({ disabled: undefined })).toBeEnabled();
 
-      expect(renderOption({id: 'foo', name: 'Foo'})).toHaveProp('isChecked', false);
-      expect(renderOption({id: 'bar', name: 'Bar'})).toHaveProp('isChecked', false);
-      expect(renderOption({id: null, name: 'Null'})).toHaveProp('isChecked', false);
+      expect(renderAndGetTrigger({ options: [] })).toBeDisabled();
+      expect(renderAndGetTrigger({ options: [], disabled: true })).toBeDisabled();
+      expect(renderAndGetTrigger({ options: [], disabled: null })).toBeDisabled();
+      expect(renderAndGetTrigger({ options: [], disabled: undefined })).toBeDisabled();
     });
 
-    it('renders enabled <NxCheckbox> for all options if component is not disabled', function () {
-      const renderOption = getShallow().renderProp('renderOption');
+    it('fires the component\'s onToggleCollapse when clicked if there are options', async function() {
+      const user = userEvent.setup(),
+          onToggleCollapseOptions = jest.fn(),
+          onToggleCollapseNoOptions = jest.fn(),
+          optionsTrigger = renderAndGetTrigger({
+            onToggleCollapse: onToggleCollapseOptions
+          }),
+          noOptionsTrigger = renderAndGetTrigger({
+            onToggleCollapse: onToggleCollapseNoOptions,
+            options: []
+          });
+
+      expect(onToggleCollapseNoOptions).not.toHaveBeenCalled();
+      await user.click(noOptionsTrigger);
+      expect(onToggleCollapseNoOptions).not.toHaveBeenCalled();
 
-      expect(renderOption({id: 'foo', name: 'Foo'})).toHaveProp('disabled', false);
-      expect(renderOption({id: 'bar', name: 'Bar'})).toHaveProp('disabled', false);
-      expect(renderOption({id: null, name: 'Null'})).toHaveProp('disabled', false);
+      expect(onToggleCollapseOptions).not.toHaveBeenCalled();
+      await user.click(optionsTrigger);
+      expect(onToggleCollapseOptions).toHaveBeenCalled();
```

Comment:
You don't really need two separate spies here.  With the clicks occurring in this order, even if the two components used the same `onToggleCollapse` callback you'd be sure that it wasn't until after the second click that it fired.

Reply:
Agree, updated here 3b5cf6e1f3e6a1c1529fc7fe5fd85aab4f8eaaaa